libcudacxx/include/cuda/std/__exception/cuda_error.h

```diff
@@ -84,17 +91,26 @@ static char* __format_cuda_error(
 class cuda_error : public ::std::runtime_error
 {
 public:
-  cuda_error(const int __status,
+  cuda_error(const __cuda_error_t __status,
              const char* __msg,
              const char* __api                    = nullptr,
              _CUDA_VSTD::source_location __loc    = _CUDA_VSTD::source_location::current(),
              __detail::__msg_storage __msg_buffer = {}) noexcept
       : ::std::runtime_error(__detail::__format_cuda_error(__msg_buffer, __status, __msg, __api, __loc))
+      , __status_(__status)
   {}
+
+  auto status() const noexcept -> __cuda_error_t
+  {
+    return __status_;
+  }
+
+private:
+  __cuda_error_t __status_;
 };
 
 [[noreturn]] _LIBCUDACXX_HIDE_FROM_ABI void __throw_cuda_error(
-  [[maybe_unused]] const int __status,
+  [[maybe_unused]] const __cuda_error_t __status,
```

Comment:
There was a question whether we should clear the cuda error here so that a user does not have to reach abck to the cuda API

Reply:
we'll do that in a separate pr.